src/allocator_thread.c

```diff
@@ -3,6 +3,7 @@
 #undef _POSIX_C_SOURCE
 #define _POSIX_C_SOURCE 200809L
 #define _DARWIN_C_SOURCE
+#define _XOPEN_SOURCE 600
```

Comment:
what for ? setting _XOPEN_SOURCE to an old version might have unwanted consequences using other C libraries (that's why i used _DARWIN_C_SOURCE, instead of _XOPEN_SOURCE)

Reply:
The original source failed to compile with error:

Reply:
see my suggestion below https://github.com/rofl0r/proxychains-ng/pull/212#discussion_r160243622